### ISSUE TITLE
filterAndTrim error -- superclass "Mnumeric" not defined

### ISSUE THREAD

**Issue by cramjaco** [benjjneb/dada2#1378]
Hello team. I've had a problem emerge on my computer with filterAndTrim. This seems to occur only in some, but not other renv environments on my system. In general, in most cases when I run filterAndTrim I get the error

```
> Error in validObject(.Object) : 
>   invalid class “SRFilterResult” object: superclass "Mnumeric" not defined in the environment of the object's class
```

which seems similar to this post, though the circumstances aren't quite the same.

https://github.com/benjjneb/dada2/issues/212

I've found that after updating to the newest version of dada2, I don't get this problem if I create a new r environment with renv, that doesn't have some of the other dependencies that I am using. For instance if I create a new environment that looks like this

```
> sessionInfo()
R version 3.6.3 (2020-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] dada2_1.16.0 Rcpp_1.0.7  

loaded via a namespace (and not attached):
 [1] plyr_1.8.6                  RColorBrewer_1.1-2         
 [3] pillar_1.6.1                compiler_3.6.3             
 [5] GenomeInfoDb_1.22.1         XVector_0.26.0             
 [7] bitops_1.0-7                tools_3.6.3                
 [9] zlibbioc_1.32.0             lattice_0.20-44            
[11] lifecycle_1.0.0             tibble_3.1.2               
[13] gtable_0.3.0                png_0.1-7                  
[15] pkgconfig_2.0.3             rlang_0.4.11               
[17] Matrix_1.3-4                DelayedArray_0.12.3        
[19] parallel_3.6.3              GenomeInfoDbData_1.2.2     
[21] stringr_1.4.0               hwriter_1.3.2              
[23] Biostrings_2.54.0           S4Vectors_0.24.4           
[25] vctrs_0.3.8                 IRanges_2.20.2             
[27] stats4_3.6.3                grid_3.6.3                 
[29] Biobase_2.46.0              glue_1.4.2                 
[31] R6_2.5.0                    jpeg_0.1-8.1               
[33] fansi_0.5.0                 BiocParallel_1.20.1        
[35] latticeExtra_0.6-29         reshape2_1.4.4             
[37] ggplot2_3.3.5               magrittr_2.0.1             
[39] Rsamtools_2.2.3             matrixStats_0.59.0         
[41] scales_1.1.1                ellipsis_0.3.2             
[43] BiocGenerics_0.32.0         GenomicRanges_1.38.0       
[45] GenomicAlignments_1.22.1    ShortRead_1.44.3           
[47] SummarizedExperiment_1.16.1 colorspace_2.0-2           
[49] renv_0.9.3                  utf8_1.2.1                 
[51] stringi_1.7.3               RCurl_1.98-1.3             
[53] RcppParallel_5.1.4          munsell_0.5.0              
[55] crayon_1.4.1   
```

I can then run this tutorial
https://benjjneb.github.io/dada2/tutorial.html

through the filter and trim step, and it works normally.

However, in my main working environment, which looks like this:

```
> sessionInfo()
R version 3.6.3 (2020-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] dada2_1.16.0 Rcpp_1.0.7  

loaded via a namespace (and not attached):
 [1] SummarizedExperiment_1.16.1 tidyselect_1.1.1           
 [3] purrr_0.3.4                 reshape2_1.4.4             
 [5] lattice_0.20-44             colorspace_2.0-2           
 [7] vctrs_0.3.8                 generics_0.1.0             
 [9] stats4_3.6.3                utf8_1.2.1                 
[11] rlang_0.4.11                pillar_1.6.1               
[13] glue_1.4.2                  DBI_1.1.1                  
[15] BiocParallel_1.20.1         BiocGenerics_0.32.0        
[17] RColorBrewer_1.1-2          plyr_1.8.6                 
[19] matrixStats_0.59.0          jpeg_0.1-8.1               
[21] GenomeInfoDbData_1.2.2      lifecycle_1.0.0            
[23] stringr_1.4.0               zlibbioc_1.32.0            
[25] Biostrings_2.54.0           munsell_0.5.0              
[27] gtable_0.3.0                hwriter_1.3.2              
[29] labeling_0.4.2              latticeExtra_0.6-29        
[31] Biobase_2.46.0              IRanges_2.20.2             
[33] GenomeInfoDb_1.22.1         parallel_3.6.3             
[35] fansi_0.5.0                 renv_0.13.2                
[37] scales_1.1.1                BiocManager_1.30.16        
[39] DelayedArray_0.12.2         S4Vectors_0.24.3           
[41] RcppParallel_5.1.4          XVector_0.26.0             
[43] ShortRead_1.44.3            farver_2.1.0               
[45] Rsamtools_2.2.3             digest_0.6.27              
[47] ggplot2_3.3.5               png_0.1-7                  
[49] stringi_1.7.3               dplyr_1.0.7                
[51] GenomicRanges_1.38.0        grid_3.6.3                 
[53] tools_3.6.3                 bitops_1.0-7               
[55] magrittr_2.0.1              RCurl_1.98-1.3             
[57] tibble_3.1.2                crayon_1.4.1               
[59] pkgconfig_2.0.3             ellipsis_0.3.2             
[61] Matrix_1.3-4                assertthat_0.2.1           
[63] R6_2.5.0                    GenomicAlignments_1.22.1   
[65] compiler_3.6.3  

```

Then when I run that same tutorial (adding the MiSeq_SOP folder to my current working directory). When I get to the step

```
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(240,160),
              maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=FALSE) # not using multithreading, otherwise same as in tutorial
```

I get the error message. 

I've upgraded all packages, with no improvement. I haven't tried updating R to version 4, since it will break a bunch of things on my system but I can try that next if you think it would help.

I also haven't tried this on another computer yet, but can if that would help. Thanks for any suggestions.

**Comment by cramjaco** [comment]
Apparenlty control return sends these out. Editing now.

**Comment by cramjaco** [comment]
This issue also looks related
https://forum.qiime2.org/t/dada2-denoise-paired-error/4025

**Comment by benjjneb** [comment]
Basically, some change in Matrix 1.3.3 broke a function dada2 was using in the ShortRead package. So you can either downgrade to Matrix 1.3.2, or use the current dada2 release where all calls to the offending code are removed.

Workaround by downgrading to Matrix 1.3.2: https://github.com/benjjneb/dada2/issues/212#issuecomment-836757295

Or fix by updating to current dada2 release 1.20: https://github.com/benjjneb/dada2/issues/212#issuecomment-840782264

**Comment by cramjaco** [comment]
Thank you for the quick response Dr. Callahan. I upgraded to dada2 v1.20 and things are indeed working now. Glad to be back on track with this project!
